optimizely/event_dispatcher.py

```diff
@@ -34,8 +34,11 @@ def dispatch_event(event):
 
     try:
       if event.http_verb == enums.HTTPVerbs.GET:
-        requests.get(event.url, params=event.params, timeout=REQUEST_TIMEOUT)
+        requests.get(event.url, params=event.params, timeout=REQUEST_TIMEOUT).raise_for_status()
       elif event.http_verb == enums.HTTPVerbs.POST:
-        requests.post(event.url, data=json.dumps(event.params), headers=event.headers, timeout=REQUEST_TIMEOUT)
+        requests.post(
+          event.url, data=json.dumps(event.params), headers=event.headers, timeout=REQUEST_TIMEOUT
+        ).raise_for_status()
+
     except request_exception.RequestException as error:
```

Comment:
It looks like this would catch any `requests.HTTPError`, which makes me wonder whether the PR is titled correctly. At the very least, I would describe it in the release notes as "Updated default event dispatcher to log HTTP errors".

@aliabbasrizvi what do you think?

Reply:
And by the way, should we consider calling the customer's error handler as well?

Or should we _actually_ raise upon certain kinds of errors? Instead of catching and logging from within the default event dispatcher.